src/test/javascript/spec/component/lecture/pdf-preview.component.spec.ts

```diff
@@ -0,0 +1,139 @@
+                numPages: 1,
+                getPage: jest.fn(() =>
+                    Promise.resolve({
+                        getViewport: jest.fn(() => ({ width: 600, height: 800, scale: 1 })),
+                        render: jest.fn(() => ({
+                            promise: Promise.resolve(),
+                        })),
+                    }),
+                ),
+            }),
+        })),
+    };
+});
+
+jest.mock('pdfjs-dist/build/pdf.worker', () => {
+    return {};
+});
+
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { ActivatedRoute } from '@angular/router';
+import { of } from 'rxjs';
+import { AttachmentService } from 'app/lecture/attachment.service';
+import { AttachmentUnitService } from 'app/lecture/lecture-unit/lecture-unit-management/attachmentUnit.service';
+import { PdfPreviewComponent } from 'app/lecture/pdf-preview/pdf-preview.component';
+import { ElementRef } from '@angular/core';
+import { AlertService } from 'app/core/util/alert.service';
+
+describe('PdfPreviewComponent', () => {
+    let component: PdfPreviewComponent;
+    let fixture: ComponentFixture<PdfPreviewComponent>;
+    let attachmentServiceMock: any;
+    let attachmentUnitServiceMock: any;
+    let routeMock: any;
+    let alertServiceMock: any;
+
+    beforeEach(async () => {
+        global.URL.createObjectURL = jest.fn().mockReturnValue('mocked_blob_url');
+        attachmentServiceMock = {
+            getAttachmentFile: jest.fn().mockReturnValue(of(new Blob([''], { type: 'application/pdf' }))),
+        };
+        attachmentUnitServiceMock = {
+            getAttachmentFile: jest.fn().mockReturnValue(of(new Blob([''], { type: 'application/pdf' }))),
+        };
+        alertServiceMock = {
+            error: jest.fn(),
+        };
+        routeMock = {
+            data: of({
+                attachment: { id: 1, name: 'Example PDF' },
+                attachmentUnit: { id: 1, name: 'Chapter 1' },
+            }),
+        };
+        alertServiceMock = {
+            addAlert: jest.fn(),
+            error: jest.fn(),
+        };
+
+        await TestBed.configureTestingModule({
+            declarations: [PdfPreviewComponent],
+            providers: [
+                { provide: ActivatedRoute, useValue: routeMock },
+                { provide: AttachmentService, useValue: attachmentServiceMock },
+                { provide: AttachmentUnitService, useValue: attachmentUnitServiceMock },
+                { provide: AlertService, useValue: alertServiceMock },
+            ],
+        }).compileComponents();
+
+        fixture = TestBed.createComponent(PdfPreviewComponent);
+        component = fixture.componentInstance;
+        component.pdfContainer = new ElementRef(document.createElement('div'));
+        component.enlargedCanvas = new ElementRef(document.createElement('canvas'));
+        fixture.detectChanges();
+    });
+
+    it('should load PDF', async () => {
```

Comment:
We don't `await` anything here, do we?

```suggestion
    it('should load PDF', () => {
```

Reply:
Should be some copy paste error, fixed